### PR TITLE
Fix initial dev setup issues

### LINKED ISSUE #242: @codecakes Outline and refine Docker setup and add documentation and Gotchas steps;
_Originally posted by @jayeclark in https://github.com/Xcov19/mycovidconnect/issues/233#issuecomment-987194703_

First timers to the project get confused when running the project. It is better to document and outline which **scopes** of the project they can run locally (for now)

Documentation is required to limit functionality and explicitly mention it:

- [x] The contributor tries to run normally with docker but need to scope out what things will work on local. For instance, trying to login won't work. For that you need to setup your own auth0 account, for web client only and register a client. This will be your own client. Their starter tutorial for this should suffice in setting up the client and using that instead of our .env vars;
- [x] react-scripts not found. Is create-react even installed locally? Try `npm install` first
- [ ] Installing with `npm install` throws error: There is a peer dependency issue with jest module that can be fixed by installing @types/jest with the right version.

Multiple PRs accpeted to close this issue;

### PR DESCRIPTION
**Pull request by jayeclark**
Signed-off-by: Jay Clark <jay@jayeclark.dev>

Fixes #242 

I was able to fix a few of the issues that had been preventing me from getting a dev environment running locally (outside of docker) for the front-end:
* Node-sass has reached end-of-life and is causing issues on new installs running newer versions of node. These issues prevent `npm i` from running successfully, which is why users may encounter the react-scripts not found error after running `npm i` - it's not clear that the install completely failed until you inspect the contents of `node_modules`. I've found on other projects that switching to sass eliminates this issue with no need for refactoring. 
* Added a .env.example file. This is helpful in making it very clear to new contributors what are the key environment variables they might need to add in order to interact with certain areas of the app. This information is available elsewhere in the repository if you poke around enough, but no need to make new contributors go searching for it.
* Added instructions to the README that make it clear that certain environment variables need to be set based on which areas of the app one wishes to work on, and that `npm install` needs to be run. 
* Miscellaneous housekeeping: Added .DS_Store to .gitignore

I do encounter a slightly long loading time (~30 seconds of the bare "loading" screen) when firing up the development server, but I as far as I can tell, that's related to either a slow network locally or a temporary issue with Auth0, and not to my changes.

<a href="https://gitpod.io/#https://github.com/Xcov19/mycovidconnect/pull/246"></a>

---
This change is [Reviewable](https://reviewable.io/reviews/xcov19/mycovidconnect/246)

### CODE REVIEW

**Comment by codecakes** [comment]
> * Node-sass has reached end-of-life and is causing issues on new installs running newer versions of node. These issues prevent `npm i` from running successfully, which is why users may encounter the react-scripts not found error after running `npm i` - it's not clear that the install completely failed until you inspect the contents of `node_modules`. I've found on other projects that switching to sass eliminates this issue with no need for refactoring.

when you say node-sass do you mean `"sass": "^1.45.1",` or node-sass as a react-script dependency?

File: src/__test__/Nearby.test.js [hunk 1]
```diff
@@ -27,7 +29,7 @@ describe('Testing Nearby component', () => {
     let wrapper = shallow(<OgNearby {...prop} />);
 
     // mocking fetch
-    fetch = jest.fn((arg) => Promise.resolve({
+    global.fetch = jest.fn((arg) => Promise.resolve({
```
Comment (codecakes) [inline src/__test__/Nearby.test.js:32]: is this necessary? or can we use another declared variable in this scope and reuse. I've not checked if this is being reused.

File: README.md [hunk 2]
```diff
@@ -71,9 +71,25 @@ Babel does the transpiling from target to es5 browser compatible code, tsc does
 
 <a name="run"></a>
 
-## 4. Running
+## 4. Developing
```
Comment (codecakes) [inline README.md]: This is good. I was thinking the following part as mentioned [here](https://github.com/jayeclark/mycovidconnect#3-installation) doesn't make sense. :
```
docker-compose -f "docker-compose.yml" up -d --build
```

If this command doesn't work, run

`PORT=9119 docker-compose up`

this is more about how to build & run it on your local. the installation section is a pre-requisite to have the necessary stack before you can run the above :point_up:

File: package.json [hunk 3]
```diff
@@ -17,13 +17,13 @@
         "eslint-plugin-jsx-a11y": "^6.3.1",
         "fibers": ">=3.1.0",
         "node-gyp": "^7.1.0",
-        "node-sass": "^5.0.0",
```
Comment (codecakes) [inline package.json:20]: i saw this now, ignore my last pr comment

**Review by codecakes — COMMENTED** [review]
looks good. I've left some remarks. See if it makes sense.

**Review by codecakes — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (jayeclark) [reply README.md]: The docker app is a bit difficult to work with as I'm not able to easily edit the .env values, and the created .env file from the docker command doesn't 'live' in my local directory... (in fact, I can't figure out where it lives or if one is even created...)

**Comment by jayeclark** [comment]
Added an additional commit clarifying the difference between sections 3 and 4. 

I'm not sure lines 36-38 in the Dockerfile actually need to be in there. It creates a .env file if one doesn't exist in the main repo, but the resulting file is not easily editable by the user (it lives in a random directory in the Docker container) so any developer wishing to set up their .env keys needs to manually create the file anyway. It's probably better/less confusing to instruct new contributors to set up their .env file prior to running Docker - this way, they have the file in the right directory and can just edit it and then start & stop their Docker container to apply any changes.

File: README.md (hunk 2 above)
Comment (codecakes) [reply README.md]: it doesn't exist. Dockerfile assumes you created it. See: https://github.com/Xcov19/mycovidconnect/blob/36962c3a2e745c46ff4da0f06a02b9d8947c38eb/Dockerfile#L38

What you could do is, add this info in documentation and ask contributors to refer to .env.example

**Comment by codecakes** [comment]
> Added an additional commit clarifying the difference between sections 3 and 4.
> 
> I'm not sure lines 36-38 in the Dockerfile actually need to be in there. It creates a .env file if one doesn't exist in the main repo, but the resulting file is not easily editable by the user (it lives in a random directory in the Docker container) so any developer wishing to set up their .env keys needs to manually create the file anyway. It's probably better/less confusing to instruct new contributors to set up their .env file prior to running Docker - this way, they have the file in the right directory and can just edit it and then start & stop their Docker container to apply any changes.

Maybe this will help >> https://github.com/Xcov19/mycovidconnect/pull/246#discussion_r777248995

**Comment by codecakes** [comment]
@jayeclark there're conflicts. update against develop and then merge

File: README.md [hunk 4]
```diff
@@ -46,6 +46,19 @@ In general follow this [Javascript Styleguide](https://google.github.io/stylegui
 
 ## 3. Installation
 
+These instructions will create a Docker container with the required stack for running and developing the project. If you have Node v14 or higher installed already, and are working on the front end, you can run the front-end locally without preparing at Docker container by following the steps in section 4.  
+#### Prepare .env keys
+
+Copy the .env.example file to create your own local .env file.
```
Comment (codecakes) [inline README.md:52]: makes sense

**Review by codecakes — APPROVED** [review]
There're some other remarks. resolve and you can merge

**Comment by codecakes** [comment]
@jayeclark can you resolve the package.json conflict and merge this branch?

**Comment by jayeclark** [comment]
Sorry, I missed your note from last month … I’ll take a look tomorrow and see if I can sort it out.

**Pull request merged**